src/main/java/org/jenkinsci/plugins/workflow/steps/EnvStep.java

```diff
@@ -150,11 +153,37 @@
                     }
                 }
                 return b.toString();
+            } else if (overrides instanceof Map) {
+                return ((Map<?, ?>) overrides).keySet()
+                    .stream()
+                    .filter(e -> e instanceof String)
+                    .map(Object::toString)
+                    .collect(Collectors.joining(", "));
             } else {
                 return null;
             }
         }
 
+        @NonNull
+        @Override
+        public Map<String, Object> customInstantiate(@NonNull Map<String, Object> arguments) {
+            Map<String, Object> r = new HashMap<>(arguments);
+            final String key = "overrides";
```

Comment:
This design forces people to write, say,

```groovy
withEnv([VAR: 'value']) {/* … */}
```

rather than the possibly more natural

```groovy
withEnv(VAR: 'value') {/* … */}
```

which would be available for anything other than an environment variable named (weirdly) `overrides`.

Reply:
Hmmm I think we need to support both as we are trying to support passing a map from scm checkout

Reply:
it's named overrides because that's how groovy passes the object.
![image](https://user-images.githubusercontent.com/1661688/73694066-a2b67180-46d7-11ea-8af1-b4aba2b3ec60.png)

Reply:
Even when I pass it like you suggested
```java
withEnv(VAR: 'value') {/* … */}
```
I still get `arguments.overrides` with whatever arguments inside.

Reply:
I'll add a test case for this